### ISSUE TITLE
DDC incompatibility

### ISSUE THREAD

**Issue by valakis** [Workiva/w_transport#277]
This small snipetts fail on ddc:
```dart
import 'dart:async';
import 'package:w_transport/browser.dart' as wt;
import 'package:w_transport/w_transport.dart' as wt;


Future main() async {
  wt.configureWTransportForBrowser();
  wt.HttpClient client = new wt.HttpClient();
  (await client.newFormRequest()
    ..uri = Uri.parse('/'))
        .post();
}
```

The problem is the `_send` method in `CommonRequest` class is returning with `Future<BaseResponse>`  and the inherited methods from the `RequestDispatchers` class is overrided like:
```dart
  @override
  Future<Response> post({dynamic body, Map<String, String> headers, Uri uri}) =>
      _send('POST', body: body, headers: headers, uri: uri);
```
The problem is `Future<BaseRespone>` is not subtype of `Future<Respone>`.

**Comment by rm-astro-wf** [comment]
It looks like you have not included any tickets in the title of this issue. Please create a ticket for this issue, or click a type to create one automatically: <a href="https://w-rmconsole.appspot.com/api/v1/issues/Workiva/w_transport/277/addTicketToTitle/Task/">Task</a>, <a href="https://w-rmconsole.appspot.com/api/v1/issues/Workiva/w_transport/277/addTicketToTitle/Bug/">Bug</a>, <a href="https://w-rmconsole.appspot.com/api/v1/issues/Workiva/w_transport/277/addTicketToTitle/Improvement/">Improvement</a>

**Comment by rm-astro-wf** [comment]
Sorry, I could not create a JIRA ticket for this issue.

I sent the following data to JIRA:
`{'issuetype': {'name': 'Bug'}, 'project': {'key': u'WP'}, 'description': "This small snipetts fail on ddc:\r\n```dart\r\nimport 'dart:async';\r\nimport 'package:w_transport/browser.dart' as wt;\r\nimport 'package:w_transport/w_transport.dart' as wt;\r\n\r\n\r\nFuture main() async {\r\n  wt.configureWTransportForBrowser();\r\n  wt.HttpClient client = new wt.HttpClient();\r\n  (await client.newFormRequest()\r\n    ..uri = Uri.parse('/'))\r\n        .post();\r\n}\r\n```\r\n\r\nThe problem is the `_send` method in `CommonRequest` class is returning with `Future<BaseResponse>`  and the inherited methods from the `RequestDispatchers` class is overrided like:\r\n```dart\r\n  @override\r\n  Future<Response> post({dynamic body, Map<String, String> headers, Uri uri}) =>\r\n      _send('POST', body: body, headers: headers, uri: uri);\r\n```\r\nThe problem is `Future<BaseRespone>` is not subtype of `Future<Respone>`.", 'components': [{'name': u'w_transport'}], 'summary': 'DDC incompatibility'}`

I received the following data from JIRA:
`{"errorMessages":[],"errors":{"customfield_10040":"Steps to Reproduce is required."}}`

If this is an issue that you cannot resolve, please reach out to RM in the Release Management chat room.

**Comment by rmconsole2-wf** [comment]
Sorry, I could not create a JIRA ticket for this issue.

I sent the following data to JIRA:
`{'issuetype': {'name': 'Bug'}, 'project': {'key': u'WP'}, 'description': "This small snipetts fail on ddc:\r\n```dart\r\nimport 'dart:async';\r\nimport 'package:w_transport/browser.dart' as wt;\r\nimport 'package:w_transport/w_transport.dart' as wt;\r\n\r\n\r\nFuture main() async {\r\n  wt.configureWTransportForBrowser();\r\n  wt.HttpClient client = new wt.HttpClient();\r\n  (await client.newFormRequest()\r\n    ..uri = Uri.parse('/'))\r\n        .post();\r\n}\r\n```\r\n\r\nThe problem is the `_send` method in `CommonRequest` class is returning with `Future<BaseResponse>`  and the inherited methods from the `RequestDispatchers` class is overrided like:\r\n```dart\r\n  @override\r\n  Future<Response> post({dynamic body, Map<String, String> headers, Uri uri}) =>\r\n      _send('POST', body: body, headers: headers, uri: uri);\r\n```\r\nThe problem is `Future<BaseRespone>` is not subtype of `Future<Respone>`.", 'components': [{'name': u'w_transport'}], 'summary': 'DDC incompatibility'}`

I received the following data from JIRA:
`{"errorMessages":[],"errors":{"customfield_10040":"Steps to Reproduce is required."}}`

If this is an issue that you cannot resolve, please reach out to RM in the Release Management chat room.

**Comment by evanweible-wf** [comment]
Thanks for reporting this @valakis, I'm working on a fix now and will have a PR up shortly.

**Comment by evanweible-wf** [comment]
https://github.com/Workiva/w_transport/pull/278

**Comment by evanweible-wf** [comment]
@valakis a fix for DDC compatibility has been released in v3.0.6
